File: Python/Hard/sliding_window/countSubArrayswithFixedBounds.py
def solution(nums, minK, maxK):
    minl = -1
    maxl = -1
    res = 0
    left = 0

    for right in range(len(nums)):
        val = nums[right]
        if val < minK or val > maxK:
            minl = -1
            maxl = -1
            left = right + 1
        else:
            if val == minK:
                minl = right
            if val == maxK:
                maxl = right
            res += max(0, min(minl, maxl) - left + 1)

    return res

File: Typescript/slidingWindow/countSubarrayswithFixedBounds.ts
var sol = function(nums: number[], minK: number, maxK: number): number {
    let minl: number = -1
    let maxl: number = -1
    let res = 0
    let left: number = 0

    for (let right = 0; right < nums.length; right ++){
        let val = nums[right]
        if (val < minK || val > maxK){
            left = right + 1
            minl = -1
            maxl  -1
        } else {
            if (right === minK) minl = right
            if (right === maxK) maxl = right
            res += Math.max(0, Math.min(minl, maxl) - left + 1)
        }
    }
    return res
}